test/integration/util/common.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	commonutil "k8s.io/minikube/pkg/util"
+)
+
+// WaitForBusyboxRunning waits until busybox pod to be running
+func WaitForBusyboxRunning(t *testing.T, namespace string, miniProfile string) error {
+	client, err := commonutil.GetClient(miniProfile)
+	if err != nil {
+		return errors.Wrap(err, "getting kubernetes client")
+	}
+	selector := labels.SelectorFromSet(labels.Set(map[string]string{"integration-test": "busybox"}))
+	return commonutil.WaitForPodsWithLabelRunning(client, namespace, selector)
+}
+
+// Retry tries the callback for a number of attempts, with a delay between attempts
+func Retry(t *testing.T, callback func() error, d time.Duration, attempts int) (err error) {
+	for i := 0; i < attempts; i++ {
+		err = callback()
+		if err == nil {
+			return nil
+		}
+		time.Sleep(d)
+	}
+	return err
+}
+
+// Retry2 tries the callback for a number of attempts, with a delay without *testing.T
+func Retry2(callback func() error, d time.Duration, attempts int) (err error) {
+	for i := 0; i < attempts; i++ {
+		err = callback()
+		if err == nil {
+			return nil
+		}
+		time.Sleep(d)
+	}
+	return err
+}
+
+// RetryX is expontential backoff retry
```

Comment:
This looks good. What do you think about replacing Retry() with this implementation and dropping Retry2()?

Reply:
I will replace all the retries in minikube in a separate PR, for better PR review